Core/scripts/dirac-install.py

```diff
@@ -1347,320 +1395,527 @@ def installLCGutils( releaseConfig ):
-                       'test $?DIRACPLAT -eq 1 || setenv DIRACPLAT `$DIRAC/scripts/dirac-platform`'] )
+        lines.extend(['# DIRAC platform',
+                      'test $?DIRACPLAT -eq 1 || setenv DIRACPLAT `$DIRAC/scripts/dirac-platform`'])
       # Add the lines required for ARC CE support
-      lines.extend( ['# ARC Computing Element',
-                     'setenv ARC_PLUGIN_PATH $DIRACLIB/arc'] )
-      lines.append( '' )
-      f = open( cshrcFile, 'w' )
-      f.write( '\n'.join( lines ) )
+      lines.extend(['# ARC Computing Element',
+                    'setenv ARC_PLUGIN_PATH $DIRACLIB/arc'])
+      lines.append('')
+      f = open(cshrcFile, 'w')
+      f.write('\n'.join(lines))
       f.close()
   except Exception, x:
-    logERROR( str( x ) )
+    logERROR(str(x))
     return False
 
   return True
 
+
 def writeDefaultConfiguration():
   instCFG = releaseConfig.getInstallationCFG()
   if not instCFG:
     return
   for opName in instCFG.getOptions():
-    instCFG.delPath( opName )
+    instCFG.delPath(opName)
 
   # filePath = os.path.join( cliParams.targetPath, "defaults-%s.cfg" % cliParams.installation )
   # Keep the default configuration file in the working directory
   filePath = "defaults-%s.cfg" % cliParams.installation
   try:
-    fd = open( filePath, "wb" )
-    fd.write( instCFG.toString() )
+    fd = open(filePath, "wb")
+    fd.write(instCFG.toString())
     fd.close()
   except Exception, excp:
-    logERROR( "Could not write %s: %s" % ( filePath, excp ) )
-  logNOTICE( "Defaults written to %s" % filePath )
+    logERROR("Could not write %s: %s" % (filePath, excp))
+  logNOTICE("Defaults written to %s" % filePath)
+
 
-def __getTerminfoLocations( defaultLocation=None ):
+def __getTerminfoLocations(defaultLocation=None):
   """returns the terminfo locations as a colon separated string"""
 
   terminfoLocations = []
   if defaultLocation:
-    terminfoLocations = [ defaultLocation ]
+    terminfoLocations = [defaultLocation]
 
-  for termpath in [ '/usr/share/terminfo', '/etc/terminfo' ]:
-    if os.path.exists( termpath ):
-      terminfoLocations.append( termpath )
+  for termpath in ['/usr/share/terminfo', '/etc/terminfo']:
+    if os.path.exists(termpath):
+      terminfoLocations.append(termpath)
 
-  return ":".join( terminfoLocations )
+  return ":".join(terminfoLocations)
+
+
+def installDiracOS(releaseConfig):
+  diracOSVersion = releaseConfig.getDiracOSVersion(cliParams.diracOSVersion)
+  if not diracOSVersion:
+    logERROR("No diracos defined")
+    return False
+
+  if cliParams.installSource:
+    tarsURL = cliParams.installSource
+  else:
+    tarsURL = releaseConfig.getTarsLocation('DIRAC')['Value']
+
+  if not downloadAndExtractTarball(tarsURL, "diracos", diracOSVersion, cache=True):
+    return False
+  logNOTICE("Fixing externals paths...")
+  fixBuildPaths()
+  logNOTICE("Running externals post install...")
+  checkPlatformAliasLink()
+  return True
+
+
+def createBashrcForDiracOS():
```

Comment:
This function and the next are for large part a repetition of existing code. Can this be refactored?

Reply:
Yes, but in some point we will drop the old one. If I split the method and use if statement, it will create a mess. This is my opinion. But I am going to do it.

Reply:
Actually I would go with Zoltan's approach on that one... We want to test things quickly, and we will probably need many iterations. So the best way not to break things is maybe to keep them apart for a bit...